xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -248,6 +248,8 @@ public static String extractXML(Node node, int start, int length)
      *   <li>3) Escape &gt; or - at the start of the comment</li>
      *   <li>4) Escape { to prevent XWiki macro syntax</li>
      *   <li>5) Add {@code \} (unescaped as {@code ""}) at the end if the last char is {@code -}</li>
+     *   <li>6) Ensure that {@code <} is always followed by {@code \} to avoid matching as HTML tag, e.g., in
```

Comment:
* I guess this works because the HTML spec says that after `<` you cannot have the `\` character (when specifying a tag)? A reference to the spec would be nice.
* And we can use `\` because `\{char}` is just an escape in HTML comments and is the same as `{char}`?

This is maybe obvious but it could be interesting to mention it as I personally didn't find it so obvious.